packages/browser/src/sessionInfo/SessionInfoManager.ts

```diff
@@ -62,60 +63,54 @@ export class SessionInfoManager
   async get(
     sessionId: string,
   ): Promise<(ISessionInfo & ISessionInternalInfo) | undefined> {
-    const isLoggedIn = await this.storageUtility.getForUser(
-      sessionId,
-      "isLoggedIn",
-      {
+    const [
+      isLoggedIn,
+      webId,
+      clientId,
+      clientSecret,
+      redirectUrl,
+      refreshToken,
+      issuer,
+      tokenType,
+    ] = await Promise.all([
+      this.storageUtility.getForUser(sessionId, "isLoggedIn", {
         secure: true,
-      },
-    );
-
-    const webId = await this.storageUtility.getForUser(sessionId, "webId", {
-      secure: true,
-    });
-
-    const clientId = await this.storageUtility.getForUser(
-      sessionId,
-      "clientId",
-      {
+      }),
+      this.storageUtility.getForUser(sessionId, "webId", {
+        secure: true,
+      }),
+      this.storageUtility.getForUser(sessionId, "clientId", {
         secure: false,
-      },
-    );
-
-    const clientSecret = await this.storageUtility.getForUser(
-      sessionId,
-      "clientSecret",
-      {
+      }),
+      this.storageUtility.getForUser(sessionId, "clientSecret", {
         secure: false,
-      },
-    );
-
-    const redirectUrl = await this.storageUtility.getForUser(
-      sessionId,
-      "redirectUrl",
-      {
+      }),
+      this.storageUtility.getForUser(sessionId, "redirectUrl", {
         secure: false,
-      },
-    );
-
-    const refreshToken = await this.storageUtility.getForUser(
-      sessionId,
-      "refreshToken",
-      {
+      }),
+      this.storageUtility.getForUser(sessionId, "refreshToken", {
         secure: true,
-      },
-    );
-
-    const issuer = await this.storageUtility.getForUser(sessionId, "issuer", {
-      secure: false,
-    });
-
-    const tokenType =
-      (await this.storageUtility.getForUser(sessionId, "tokenType", {
+      }),
+      this.storageUtility.getForUser(sessionId, "issuer", {
+        secure: false,
+      }),
+      this.storageUtility.getForUser(sessionId, "tokenType", {
         secure: false,
-      })) ?? "DPoP";
+      }),
+    ]);
+
+    if (typeof redirectUrl === "string" && !isValidRedirectUrl(redirectUrl)) {
```

Comment:
Potentially, but it would become harder for the library to pick that up. One issue is that the failure happens out of the domain of the client, so the client is not informed that the OpenID Provider errors on their `redirect_url`, and there is no way for the client to look up what redirect URL has been registered by the client at dynamic registration. 

I'm not even sure how the bad redirect URL gets in storage in the first place, but I suspect that using `window.location.href` plays a role. In that case, a likely cause of failure is picking up the OpenID query params (which is the observed cause of the bug in the reproduced issues).

If what you describe is indeed an issue, we'll need to figure out how to prevent it from happening, but for the time being I think it is low probabllity enough that we don't work on a fix for the time being.